### PR TITLE
Clang format demo

### PR DESCRIPTION
**Pull request by janisozaur**
cf. #4836

### CODE REVIEW

**Comment by Broxzier** [comment]
While I absolutely love the idea of having an automatic code formatter for this project, there are some cases where formatting isn't nice, for example in how the window widgets are initialized in C arrays. Wrapping these cases within `// clang-format off` and `// clang-format on` could solve this. I'm personally not against adding those, because these arrays will be replaced at some point anyway.

**Comment by janisozaur** [comment]
How fun, this actually revealed some bugs.

1. We don't include all necessary headers in some files
2. Travis reports following:
```
In file included from /usr/include/stdlib.h:959:0,
                 from /home/travis/build/OpenRCT2/OpenRCT2/src/platform/../rct2.h:36,
                 from /home/travis/build/OpenRCT2/OpenRCT2/src/platform/../common.h:21,
                 from /home/travis/build/OpenRCT2/OpenRCT2/src/platform/../config.h:20,
                 from /home/travis/build/OpenRCT2/OpenRCT2/src/platform/posix.c:19:
In function ‘realpath’,
    inlined from ‘platform_resolve_openrct_data_path’ at /home/travis/build/OpenRCT2/OpenRCT2/src/platform/posix.c:749:21:
/usr/include/x86_64-linux-gnu/bits/stdlib.h:43:2: error: call to ‘__realpath_chk_warn’ declared with attribute warning: second argument of realpath must be either NULL or at least PATH_MAX bytes long buffer [-Werror]
  return __realpath_chk_warn (__name, __resolved, __bos (__resolved));
  ^
```

File: src/ScenarioRepository.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#pragma region Copyright (c) 2014-2016 OpenRCT2 Developers
+#pragma region Copyright(c) 2014 - 2016 OpenRCT2 Developers
```
Comment (Gymnasiast) [inline src/ScenarioRepository.cpp]: Should it autoformat this line? Looks a bit strange.

File: src/ScenarioRepository.cpp [hunk 2]
```diff
@@ -14,37 +14,40 @@
...
+#include <memory>
+#include <vector>
 
-extern "C"
-{
-    #include "config.h"
-    #include "localisation/localisation.h"
-    #include "scenario.h"
+extern "C" {
+#include "config.h"
```
Comment (Gymnasiast) [inline src/ScenarioRepository.cpp]: This looks ugly.

**Pull request closed without merging**